### PR TITLE
Feat: Add Dash Core 0.18 Support

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

This PR will add Core 18 support to Dash Wallet.

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/build.gradle [hunk 1]
```diff
@@ -107,6 +107,9 @@ dependencies {
     androidTestImplementation('androidx.test.espresso:espresso-core:3.3.0', {
         exclude group: 'com.android.support', module: 'support-annotations'
     })
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'com.google.guava:guava:29.0-android'
+    implementation "org.slf4j:slf4j-api:$slf4jVersion"
```
Comment (HashEngineering) [inline common/build.gradle:112]: This is required because DashJ 0.18 which is built by gradle instead of maven is not built as a bundle that includes all of its dependencies.

Though the kotlin lib is not used by DashJ.

File: common/src/main/java/org/dash/wallet/common/ui/CurrencyAmountView.java [hunk 2]
```diff
@@ -34,6 +32,8 @@
 import android.graphics.drawable.Drawable;
 import android.os.Bundle;
 import android.os.Parcelable;
+
+import androidx.annotation.Nullable;
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/ui/CurrencyAmountView.java:36]: we don't need this change.

File: common/src/main/java/org/dash/wallet/common/ui/OffsetDialogFragment.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import org.dash.wallet.common.R
+
+
+open class OffsetDialogFragment<T: ViewGroup> : BottomSheetDialogFragment() {
+    @DrawableRes protected open val background: Int = R.drawable.white_background_rounded
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        dialog?.setOnShowListener { dialog ->
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/ui/OffsetDialogFragment.kt:38]: I think this file is obsolete, must have been leftover after a merge.

File: integrations/uphold/src/main/java/org/dash/wallet/integration/uphold/data/UpholdApiException.java [hunk 4]
```diff
@@ -43,6 +43,10 @@ public UpholdApiException(Response response) {
         }
     }
 
+    public UpholdApiException(int httpStatusCode) {
+        this.httpStatusCode = httpStatusCode;
+    }
+
```
Comment (HashEngineering) [inline integrations/uphold/src/main/java/org/dash/wallet/integration/uphold/data/UpholdApiException.java:49]: why is this here?  we don't need it for Core 18.

File: integrations/uphold/build.gradle [hunk 5]
```diff
@@ -38,6 +38,7 @@ dependencies {
     implementation 'androidx.localbroadcastmanager:localbroadcastmanager:1.0.0'
     implementation "org.dashj:dashj-core:$dashjVersion"
     implementation 'com.scottyab:secure-preferences-lib:0.1.7'
+    implementation 'org.slf4j:slf4j-api:1.7.30'
```
Comment (HashEngineering) [inline integrations/uphold/build.gradle:41]: added because dashj doesn't include it.

File: integrations/uphold/src/test/java/org/dash/wallet/integration/uphold/UpholdErrorTest.kt [hunk 6]
```diff
@@ -1,302 +0,0 @@
...
-import org.dash.wallet.integration.uphold.data.UpholdApiException
-import org.junit.Test
-import org.junit.Assert.*
-import org.mockito.ArgumentMatchers.anyString
-import org.mockito.kotlin.*
-
-class UpholdErrorsTest {
-
-    @Test
-    fun error403Test() {
```
Comment (HashEngineering) [inline integrations/uphold/src/test/java/org/dash/wallet/integration/uphold/UpholdErrorTest.kt:13]: not sure why this file is deleted, since it is unrelated to dashj 0.18

File: common/src/main/java/org/dash/wallet/common/ui/OffsetDialogFragment.kt (hunk 3 above)
Comment (Syn-McJ) [reply common/src/main/java/org/dash/wallet/common/ui/OffsetDialogFragment.kt:38]: Yes the `OffsetDialogFragment` is located in `ui/dialogs` now

File: common/src/main/java/org/dash/wallet/common/ui/CurrencyAmountView.java (hunk 2 above)
Comment (HashEngineering) [reply common/src/main/java/org/dash/wallet/common/ui/CurrencyAmountView.java:36]: this might be another think left out of dashj.

File: wallet/src/de/schildbach/wallet/ui/WalletUri.java [hunk 7]
```diff
@@ -153,9 +153,7 @@ public BitcoinURI toBitcoinUri() throws BitcoinURIParseException {
         Coin amount = getAmount();
         String sender = getSender();
         String bitcoinUri = BitcoinURI.convertToBitcoinURI(address, amount, sender, null);
-        if (useInstantSend() || forceInstantSend()) {
-            bitcoinUri += "&" + BitcoinURI.FIELD_INSTANTSEND + "=1";
-        }
+
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/WalletUri.java:156]: The InstantSend field is obsolete.  This was from a time when instantSend had to be requested and there was an extra fee for using it.

File: integrations/uphold/src/test/java/org/dash/wallet/integration/uphold/UpholdErrorTest.kt (hunk 6 above)
Comment (HashEngineering) [reply integrations/uphold/src/test/java/org/dash/wallet/integration/uphold/UpholdErrorTest.kt]: fixed

**Review by Syn-McJ — APPROVED** [review]
Looks great

**Comment by eloyesp** [comment]
Hi, as the f-droid version is not being maintained anymore and this fix will not get there, what are the expected issues with users of that version now? What are the compatibility issues being addressed here?
